NEWS.md

```diff
@@ -1,5 +1,7 @@
 # stringr (development version)
 
+* Update `str_starts()` and `str_ends()` functions so they honor regex "|" operator more intuitively. (@carlganz)
```

Comment:
I’d phrase this more generally - it’s about operator precedence in generally, not just `|`. Otherwise, looks great.